icu4j/main/classes/core/src/com/ibm/icu/number/NumberPropertyMapper.java

```diff
@@ -275,8 +275,15 @@ public static MacroProps oldToNew(
                     // Patterns like "#.##E0" (no zeros in the mantissa), which mean round to maxFrac+1
                     macros.precision = Precision.constructSignificant(1, maxFrac_ + 1).withMode(mathContext);
                 } else {
-                    // All other scientific patterns, which mean round to minInt+maxFrac
-                    macros.precision = Precision.constructSignificant(minInt_ + minFrac_, minInt_ + maxFrac_)
+                    int maxSig_ = minInt_ + maxFrac_;
+                    // Bug #20058: if maxInt_ > minInt_ > 1, then minInt_ should be 1.
```

Comment:
Similar logic is performed a few lines earlier, line 248.

Does it work if you make `minFrac_` and `maxFrac_` but not `minInt_` and `maxInt_`, and instead use `minInt` and `maxInt` directly?